pkg/playground/react-patterns.js

```diff
@@ -40,33 +40,41 @@ import { showMultiTypeaheadDemo } from "./react-demo-multi-typeahead.jsx";
 
 let lastAction = "";
 
-const onDialogStandardClicked = function(mode) {
+const onDialogStandardClicked = function(mode, progress_cb) {
     lastAction = mode;
-    const dfd = cockpit.defer();
-    dfd.notify("Starting something long");
+    let myResolve, myReject;
+    const promise = new Promise((resolve, reject) => {
+        myResolve = resolve;
+        myReject = reject;
+    });
+
+    cockpit.assert(myResolve !== undefined);
+    cockpit.assert(myReject !== undefined);
+
+    progress_cb("Starting something long");
     if (mode == 'steps') {
+        const cancel = function() {
+            window.clearTimeout(interval);
+            progress_cb("Canceling");
+            window.setTimeout(function() {
+                myReject("Action canceled");
+            }, 1000);
```

Comment:
These 6 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-22186-804e4a75-20250708-145413-fedora-42-devel/Coverage/lcov/github-pr.diff.gcov.html)